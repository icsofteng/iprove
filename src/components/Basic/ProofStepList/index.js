import React from 'react'
import ProofStep from '../ProofStep'
import RulePlaceholder from '../RulePlaceholder'
import { is_step } from '../../../utils'

const ProofStepList = (props) => {
  return (
    <React.Fragment>
      {
        props.steps.filter(is_step).map((step, id) =>
          <ProofStep key={"step"+id} step={step} index={id} showDependencies={props.showDependencies} offset={props.start} type={props.type} z3={props.z3[id]} />
        )
      }
      <div className="proof-line">
        {
          ((props.type === 'goal' && props.steps.filter(is_step).length === 0) || props.type !== 'goal') &&
            <RulePlaceholder wide path={[props.type, props.steps.filter(is_step).length]} />
        }
        <div className="proof-feedback"></div>
      </div>
    </React.Fragment>
  )
}

export default ProofStepList

