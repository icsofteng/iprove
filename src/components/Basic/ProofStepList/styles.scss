.rulePlaceholder {
  padding: 20px;
  border: 2px dashed #666;
  color: #666;

  &:global(.drop-target) {
    background:yellow;
  }
}