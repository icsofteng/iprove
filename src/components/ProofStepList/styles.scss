.steps {
  padding: 25px;
}
.rulePlaceholder {
  padding: 20px;
  border: 2px dashed #666;
  color: #666;

  &:global(.drop-target) {
    background:yellow;
  }
}
.feedback {
  display: block;
  font-weight: bold;
  padding: 10px;
  margin-bottom: 15px;
  font-size: 12px;
  background: #148414;

  &.error {
    background:#9a4646;
  }
}