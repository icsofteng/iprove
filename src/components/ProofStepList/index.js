import React from 'react'
import ProofStep from '../ProofStep'
import styles from './styles.scss'
import RulePlaceholder from '../RulePlaceholder'

const ProofStepList = (props) =>
  <div className={styles.steps}>
    {
      props.steps.map((rule, index) =>
        <ProofStep key={"step"+index} rule={rule} index={index} />
      )
    }
    <RulePlaceholder wide path={[props.steps.length]} />
  </div>

export default ProofStepList

