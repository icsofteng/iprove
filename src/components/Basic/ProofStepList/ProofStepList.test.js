import React from 'react'
import ProofStepList from './'

const ast = {type: "binary", symbol: "implies", lhs: {type:"literal", value:"p"}, rhs:  {type:"literal", value:"q"}}
const ast2 = {type: "literal", value: "p"}
const step = { dependencies: [], ast }
const step2 = { dependencies: [], ast2 }
const steps = [step, step2]
const z3State = ["unsat"]
const wrapper = shallow(<ProofStepList z3={z3State} steps={steps} start={0} type="givens" />)

test('Snapshot test', () => {
  expect(wrapper).toMatchSnapshot()
})

