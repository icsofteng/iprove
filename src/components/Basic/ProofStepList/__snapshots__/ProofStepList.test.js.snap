// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<Fragment>
  <ProofStep
    index={0}
    key="step0"
    offset={0}
    step={
      Object {
        "ast": Object {
          "lhs": Object {
            "type": "literal",
            "value": "p",
          },
          "rhs": Object {
            "type": "literal",
            "value": "q",
          },
          "symbol": "implies",
          "type": "binary",
        },
        "dependencies": Array [],
      }
    }
    type="givens"
    z3="unsat"
  />
  <div
    className="proof-line"
  >
    <RulePlaceholder
      path={
        Array [
          "givens",
          1,
        ]
      }
      wide={true}
    />
    <div
      className="proof-feedback"
    />
  </div>
</Fragment>
`;
